test/Kestrel.InMemory.FunctionalTests/Http2/Http2TimeoutTests.cs

```diff
@@ -203,15 +203,16 @@ public async Task AbortedStream_ResetsAndDrainsRequest_RefusesFramesAfterCooldow
             };
             await InitializeConnectionAsync(_appAbort);
 
+            // Prevent the test from advancing mockSystemClock.UtcNow until the drain timeout is set.
+            var utcNowAccessedTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            mockSystemClock.OnUtcNowGetterCalled = () => utcNowAccessedTcs.TrySetResult(null);
+
             await StartStreamAsync(1, headers, endStream: false);
 
             await WaitForStreamErrorAsync(1, Http2ErrorCode.INTERNAL_ERROR, "The connection was aborted by the application.");
 
-            // There's a race when the appfunc is exiting about how soon it unregisters the stream.
```

Comment:
@Tratcher Do you think the other tests that use this pattern should be updated?

Reply:
Yes